junebug/channel.py

```diff
@@ -0,0 +1,149 @@
+import collections
+from copy import deepcopy
+import json
+import uuid
+from twisted.internet.defer import inlineCallbacks, returnValue
+from twisted.web import http
+from vumi.service import WorkerCreator
+from vumi.servicemaker import VumiOptions
+
+from junebug.error import JunebugError
+
+
+class ChannelNotFound(JunebugError):
+    '''Raised when a channel's data cannot be found.'''
+    name = 'ChannelNotFound'
+    description = 'channel not found'
+    code = http.NOT_FOUND
+
+
+class InvalidChannelType(JunebugError):
+    '''Raised when an invalid channel type is specified'''
+    name = 'InvalidChannelType',
+    description = 'invalid channel type'
+    code = http.BAD_REQUEST
+
+
+transports = {
+    'telnet': 'vumi.transports.telnet.TelnetServerTransport',
+    'xmpp': 'vumi.transports.xmpp.XMPPTransport',
+}
```

Comment:
I'm not sure what you mean by configurable? My idea was to add them individually like this, and them document them in the Junebug documentation. It means that you don't have to have `vumi.transports....` in your API request, and also documenting them separately means that we can document what each `config` thing it needs is.

I see that adding them as dependancies might be an issue, but if we're planning to release this as a debian package, wouldn't we need to include them all anyway?